### ISSUE TITLE
The users should be able to mark a step as "in progress"

### ISSUE THREAD

**Issue by kernixski** [kullervo16/checklist#70]
The ON_HOLD state should be replaced by this one.

The user should be able to mark a step as IN_PROGRESS to prevent the previous step to be reopen.  It is also handy to know which steps are in progress in a checklist.

**Comment by kernixski** [comment]
ON_HOLD -> IN_PROGRESS pushed

**Comment by kullervo16** [comment]
Just replacing ON_HOLD by IN_PROGRESS breaks the startup when there are templates that have ON_HOLD in their persistence... So I added a bit of code that converts in when reading from persistence.

**Comment by kernixski** [comment]
Right. :-)

I also plan to refactor the checklist and template loading a bit.  I plan to allow multiple conditions for a step and the structure of the files will be updated.  The loader will be able to manage multiple versions of the files structure and will convert it to the latest version.  But I will create a new issue for this later. ;-)

**Comment by kernixski** [comment]
Blocked by #80
